addons/map/XEH_postInitClient.sqf

```diff
@@ -65,15 +65,22 @@ GVAR(flashlights) = createHashMap;
 // hide clock on map if player has no watch
 GVAR(hasWatch) = true;
 
-["loadout", {
-    params ["_unit"];
-    if (isNull _unit) exitWith {
+[QGVAR(slotItemChanged), "SlotItemChanged", {
+    params ["", "_item", "_slot", "_assign"];
+
+    if (_slot != TYPE_WATCH) exitWith {};
+
+    GVAR(hasWatch) = _assign && {_item isKindOf ["ItemWatch", configFile >> "CfgWeapons"]};
+}] call EFUNC(common,addPlayerEH);
+
+["unit", {
+    params ["_newPlayer"];
+
+    if (isNull _newPlayer) exitWith {
```

Comment:
Need to find out why this check is here and see if `EFUNC(common,addPlayerEH)` needs to take virtual units into account because of that.